java/src/main/native/src/CudaJni.cpp

```diff
@@ -195,6 +195,32 @@ JNIEXPORT jint JNICALL Java_ai_rapids_cudf_Cuda_getNativeComputeMode(JNIEnv *env
   CATCH_STD(env, -2);
 }
 
+namespace {
+
+jint cuda_device_get_attribute_impl(JNIEnv *env, cudaDeviceAttr const &attribute_label) {
+  try {
+    cudf::jni::auto_set_device(env);
+    int device;
+    JNI_CUDA_TRY(env, -2, ::cudaGetDevice(&device));
+    int attribute_value;
+    JNI_CUDA_TRY(env, -2, ::cudaDeviceGetAttribute(&attribute_value, attribute_label, device));
+    return attribute_value;
+  }
+  CATCH_STD(env, -2);
+}
+
+} // namespace
+
+JNIEXPORT jint JNICALL Java_ai_rapids_cudf_Cuda_getCurrentComputeCapabilityMajor(JNIEnv *env,
+                                                                                 jclass) {
+  return cuda_device_get_attribute_impl(env, ::cudaDevAttrComputeCapabilityMajor);
+}
+
+JNIEXPORT jint JNICALL Java_ai_rapids_cudf_Cuda_getCurrentComputeCapabilityMinor(JNIEnv *env,
```

Comment:
Only quibble:  If we only need to check for <= Pascal, do we need minor version here?  I have no objections to this in principle though.

Reply:
Agreed. Not needed at the moment. This is in place just in case we've missed something, or need more fine tuning from the plugin due to unforeseen circumstances. 

I'M loath to re-re-release the CUDF in that unlikely event.